### PR TITLE
Defines the TenantManager class and its underlying methods.

### PR DESCRIPTION
**Pull request by bojeil-google**
Adds unit tests for this new class. Unit tests were copied from the
exising auth.spec.ts file.

All deprecated methods will be removed from the Auth class in a follow
up PR.

### CODE REVIEW

File: test/unit/auth/tenant-manager.spec.ts [hunk 1]
```diff
@@ -0,0 +1,590 @@
...
+      expect(tenantAwareAuth1.tenantId).to.equal('tenantId1');
+      expect(tenantAwareAuth2.tenantId).to.equal('tenantId2');
+    });
+  });
+
+  describe('getTenant()', () => {
+    const tenantId = 'tenant_id';
+    const serverResponse: TenantServerResponse = {
+      name: 'projects/project_id/tenants/tenant_id',
+      displayName: 'TENANT_DISPLAY_NAME',
```
Comment (wuyanna) [inline test/unit/auth/tenant-manager.spec.ts]: Use only letters, digits and hyphens to to be consistent with requirements

Comment (bojeil-google) [reply test/unit/auth/tenant-manager.spec.ts]: Done.

File: test/unit/auth/tenant-manager.spec.ts [hunk 2]
```diff
@@ -0,0 +1,590 @@
...
+        .then(() => {
+          throw new Error('Unexpected success');
+        })
+        .catch((error) => {
+          expect(error).to.have.property('code', 'auth/invalid-page-token');
+        });
+    });
+
+    it('should be rejected given an invalid max result', () => {
+      const invalidResults = 5000;
```
Comment (hiranya911) [inline test/unit/auth/tenant-manager.spec.ts]: I suppose 1000 is the allowed max. I'd mention that in the test description or change the assignment to something like:

```
const moreThanMax = 1000 + 1;
```

File: test/unit/auth/tenant-manager.spec.ts [hunk 3]
```diff
@@ -0,0 +1,590 @@
...
+  describe('createTenant()', () => {
+    const tenantId = 'tenant_id';
+    const tenantOptions: TenantOptions = {
+      displayName: 'TENANT-DISPLAY-NAME',
+      emailSignInConfig: {
+        enabled: true,
+        passwordRequired: true,
+      },
+    };
+    const serverResponse: TenantServerResponse = {
```
[the remark this answers is not in the document]
Comment (bojeil-google) [reply test/unit/auth/tenant-manager.spec.ts]: done

**Comment by bojeil-google** [comment]
Thanks for the quick review!

File: test/unit/auth/tenant-manager.spec.ts [hunk 4]
```diff
@@ -0,0 +1,590 @@
...
+      });
+    });
+
+    it('should return a TenantAwareAuth with the expected tenant ID', () => {
+      expect(tenantManager.authForTenant(TENANT_ID).tenantId).to.equal(TENANT_ID);
+    });
+
+    it('should return a TenantAwareAuth with read-only tenant ID', () => {
+      expect(() => {
+        (tenantManager.authForTenant(TENANT_ID) as any).tenantId = 'OTHER_TENANT_ID';
```
[the remark this answers is not in the document]
Comment (bojeil-google) [reply test/unit/auth/tenant-manager.spec.ts]: done

File: test/unit/auth/tenant-manager.spec.ts [hunk 5]
```diff
@@ -0,0 +1,590 @@
...
+          // Confirm underlying API called with expected parameters.
+          expect(stub).to.have.been.calledOnce.and.calledWith(tenantId);
+          // Confirm expected error returned.
+          expect(error).to.equal(expectedError);
+        });
+    });
+  });
+
+  describe('createTenant()', () => {
+    const tenantId = 'tenant_id';
```
[the remark this answers is not in the document]
Comment (bojeil-google) [reply test/unit/auth/tenant-manager.spec.ts]: done

File: test/unit/auth/tenant-manager.spec.ts [hunk 6]
```diff
@@ -0,0 +1,590 @@
...
+    const tenantOptions: TenantOptions = {
+      displayName: 'TENANT_DISPLAY_NAME',
```
[the remark this answers is not in the document]
Comment (bojeil-google) [reply test/unit/auth/tenant-manager.spec.ts]: done

File: test/unit/auth/tenant-manager.spec.ts [hunk 7]
```diff
@@ -0,0 +1,590 @@
...
+      emailSignInConfig: {
+        enabled: true,
+        passwordRequired: true,
+      },
+    };
+    const serverResponse: TenantServerResponse = {
+      name: 'projects/project_id/tenants/tenant_id',
+      displayName: 'TENANT_DISPLAY_NAME',
```
[the remark this answers is not in the document]
Comment (bojeil-google) [reply test/unit/auth/tenant-manager.spec.ts]: done

File: test/unit/auth/tenant-manager.spec.ts [hunk 8]
```diff
@@ -0,0 +1,590 @@
...
+          // Confirm underlying API called with expected parameters.
+          expect(createTenantStub).to.have.been.calledOnce.and.calledWith(tenantOptions);
+          // Confirm expected error returned.
+          expect(error).to.equal(expectedError);
+        });
+    });
+  });
+
+  describe('updateTenant()', () => {
+    const tenantId = 'tenant_id';
```
[the remark this answers is not in the document]
Comment (bojeil-google) [reply test/unit/auth/tenant-manager.spec.ts]: done

File: test/unit/auth/tenant-manager.spec.ts [hunk 9]
```diff
@@ -0,0 +1,590 @@
...
+    const tenantOptions: TenantOptions = {
+      displayName: 'TENANT_DISPLAY_NAME',
+      emailSignInConfig: {
+        enabled: true,
+        passwordRequired: true,
+      },
+    };
+    const serverResponse: TenantServerResponse = {
+      name: 'projects/project_id/tenants/tenant_id',
```
[the remark this answers is not in the document]
Comment (bojeil-google) [reply test/unit/auth/tenant-manager.spec.ts]: done

**Pull request merged**